activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1105,6 +1105,34 @@ def uniq!(name)
       self
     end
 
+    # Excludes the specified record from the resulting relation. For example:
+    #
+    #   Post.excludes(post)
+    #   # SELECT "posts".* FROM "posts" WHERE "posts"."id" != 1
+    #
+    # This can also be called on associations:
+    #
+    #   post = Post.find(1)
+    #   comment = Comment.find(2)
+    #   post.comments.excludes(comment)
+    #   # SELECT "comments".* FROM "comments" WHERE "comments"."post_id" = 1 AND "comments"."id" != 2
+    #
+    # This is short-hand for <tt>Post.where.not(id: post.id)</tt>.
+    #
+    # An <tt>ArgumentError</tt> will be raised if the specified record is not an instance
+    # of the same model that the relation is scoping.
+    def excludes(record)
+      if record.is_a?(klass)
```

Comment:
To follow along with `Enumerable#excluding` I do think we want to support passing a separate collection of records. Could do:

```suggestion
    def excludes(*records)
      records.flatten!(1).select!(klass)
      if records.any?
```